tests/LinAlg/vectorTest.hpp

```diff
@@ -1527,19 +1537,151 @@ class VectorTests : public TestBase
     return reduceReturn(fail, &x);
   }
 
-protected:
-  // Interface to methods specific to vector implementation
-  virtual void setLocalElement(hiop::hiopVector* x, local_ordinal_type i, real_type val) = 0;
-  virtual real_type getLocalElement(const hiop::hiopVector* x, local_ordinal_type i) = 0;
-  virtual local_ordinal_type getLocalSize(const hiop::hiopVector* x) = 0;
-  virtual real_type* getLocalData(hiop::hiopVector* x) = 0;
-  virtual int verifyAnswer(hiop::hiopVector* x, real_type answer) = 0;
+  /// Method to set vector _x_ element _i_ to _value_.
+  void setLocalElement(hiop::hiopVector* x, local_ordinal_type i, real_type val)
+  {
+    x->copyFromDev();
+    real_type *xdat = x->local_data_host();
+    xdat[i] = val;
+    x->copyToDev();
+  }
+  /// Returns element _i_ of vector _x_.
+  real_type getLocalElement(const hiop::hiopVector* x, local_ordinal_type i)
+  {
+    x->copyFromDev();
+    return x->local_data_host_const()[i];
+  }
```

Comment:
I suggest keeping these in a implementations specific file. While this might work for the test cases we have now, it is a short-sighted solution.